src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php

```diff
@@ -0,0 +1,183 @@
+
+                return $this->redirectToRoute('admin_manufacturers_index');
+            }
+        } catch (CoreException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/add.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+            'manufacturerForm' => $manufacturerForm->createView(),
+        ]);
+    }
+
+    /**
+     * Show & process manufacturer editing.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update'], request.get('_legacy_controller'))"
+     * )
+     *
+     * @param int $manufacturerId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(Request $request, $manufacturerId)
+    {
+        try {
+            $manufacturerForm = $this->getFormBuilder()->getFormFor((int) $manufacturerId);
+            $manufacturerForm->handleRequest($request);
+
+            $result = $this->getFormHandler()->handleFor((int) $manufacturerId, $manufacturerForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_manufacturers_index');
+            }
+        } catch (CoreException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            if ($e instanceof ManufacturerNotFoundException) {
+                return $this->redirectToRoute('admin_manufacturers_index');
+            }
+        }
+
+        /** @var EditableManufacturer $editableManufacturer */
+        $editableManufacturer = $this->getQueryBus()->handle(new GetManufacturerForEditing((int) $manufacturerId));
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/edit.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+            'manufacturerForm' => $manufacturerForm->createView(),
+            'manufacturerName' => $editableManufacturer->getName(),
+            'logoImage' => $editableManufacturer->getLogoImage(),
+        ]);
+    }
+
+    /**
+     * Provides error messages for exceptions
+     *
+     * @return array
+     */
+    private function getErrorMessages()
+    {
+        $iniConfig = $this->get('prestashop.core.configuration.ini_configuration');
+
+        return [
+            ManufacturerNotFoundException::class => $this->trans(
+                'The object cannot be loaded (or found)',
+                'Admin.Notifications.Error'
+            ),
+            MemoryLimitException::class => $this->trans(
```

Comment:
I see here that you have tried to find all possible failures from the code being used in the controller Actions in order to translate them. Is it right ?

I never thought about that but it actually make sense 🤔 wdyt @sarjon @eternoendless ?

Reply:
not sure, but i think i did that as well since hitting memory limit is "expected" result in some cases. As far as I remember, it comes from legacy image uploading.

Reply:
Its for image uploading, I'm sure that user must be notified when trying to upload too big file or wrong format. Those exceptions comes from abstract classes that aren't in same domain level as i remember so I'm catching CoreException